src/wallet/rpcwallet.cpp

```diff
@@ -2582,6 +2584,236 @@ UniValue fundrawtransaction(const JSONRPCRequest& request)
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Transaction is not BIP 125 replaceable");
+
+    if (wtx.mapValue.count("replaced_by_txid"))
+        throw JSONRPCError(RPC_INVALID_REQUEST, strprintf("Cannot bump transaction %s which was already bumped by transaction %s", hash.ToString(), wtx.mapValue.at("replaced_by_txid")));
+
+    // check that original tx consists entirely of our inputs
+    // if not, we can't bump the fee, because the wallet has no way of knowing the value of the other inputs (thus the fee)
+    if (!pwalletMain->IsAllFromMe(wtx, ISMINE_SPENDABLE))
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Transaction contains inputs that don't belong to this wallet");
+
+    // figure out which output was change
+    // if there was no change output or multiple change outputs, fail
+    int nOutput = -1;
+    for (size_t i = 0; i < wtx.tx->vout.size(); ++i) {
+        if (pwalletMain->IsChange(wtx.tx->vout[i])) {
+            if (nOutput != -1)
+                throw JSONRPCError(RPC_MISC_ERROR, "Transaction has multiple change outputs");
+            nOutput = i;
+        }
+    }
+    if (nOutput == -1)
+        throw JSONRPCError(RPC_MISC_ERROR, "Transaction does not have a change output");
+
+    // optional parameters
+    int newConfirmTarget = nTxConfirmTarget;
+    CAmount totalFee = 0;
+    bool replaceable = true;
+    if (request.params.size() > 1) {
+        UniValue options = request.params[1];
+        if (options.size() > 2)
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Too many optional parameters");
+        RPCTypeCheckObj(options,
+            {
+                {"confTarget", UniValueType(UniValue::VNUM)},
+                {"totalFee", UniValueType(UniValue::VNUM)},
+                {"replaceable", UniValueType(UniValue::VBOOL)},
+            },
+            true, true);
+
+        if (options.exists("confTarget") && options.exists("totalFee")) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "confTarget and totalFee options should not both be set. Please provide either a confirmation target for fee estimation or an explicit total fee for the transaction.");
+        } else if (options.exists("confTarget")) {
+            newConfirmTarget = options["confTarget"].get_int();
+            if (newConfirmTarget <= 0) // upper-bound will be checked by estimatefee/smartfee
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid confTarget (cannot be <= 0)");
+        } else if (options.exists("totalFee")) {
+            totalFee = options["totalFee"].get_int();
+            if (totalFee <= 0)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid totalFee (cannot be <= 0)");
+            else if (totalFee > maxTxFee)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid totalFee (cannot be higher than maxTxFee)");
+        }
+
+        if (options.exists("replaceable") && !options["replaceable"].get_bool()) {
+            replaceable = false;
```

Comment:
Prefer to just do:

```C++
if (options.exists("replaceable")) {
    replaceable = options["replaceable"].get_bool();
}
```

So that if the default changes, this still works right.